### PR TITLE
Switch from elasticearch-maven-plugin to docker-maven-plugin to start Elasticsearch for ITs

### LINKED ISSUE #11597: Avoid starting Elasticsearch with UseConcMarkSweepGC (on the primary JVM)
**Description**
For purposes of integration tests we're starting Elasticsearch nodes, which use their default configuration which sets `UseConcMarkSweepGC`.
When GraalVM integration tests are run on Quarkus, the GraalVM master is the primary JVM being used across all tests; this JVM doesn't support the chosen GC.

**Implementation ideas**
It's not particularly important which GC is being used, as we just need to be able to connect to the Elasticsearch service for purposes of integration testing. 
We could either switch GC in the configuration files, or switch to container images - which is consistent to how we run other such services.
It's also not a goal to actually run Elasticsearch with the GraalVM JVM; while this could be interesting, we're unlikely to have bandwith to diagnose possible trouble in that area too, so running ES via a stable container image is probably the best option.

We need this fix to help with https://github.com/oracle/graal/issues/2782 cc/ @zakkak

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #11597 

cc @Sanne @gsmet

### CODE REVIEW

**Review by Sanne — CHANGES_REQUESTED** [review]
Looks good, but in other ITs we make sure to only expect docker if the docker profile is activated. Could you do the same?

Also I think we should adapt the tasks in `.github/` actions to pre-start a single ES container for the group. Not sure if that's easy to adapt? Feel free to spin this off as a separate task, as it would be less urgent.

**Comment by yrodiere** [comment]
@Sanne I think I did what you wanted.

**Comment by yrodiere** [comment]
The build failed, but the failures seem unrelated, and likely transient. I just force-pushed to trigger another build.

File: integration-tests/elasticsearch-rest-client/pom.xml [hunk 1]
```diff
@@ -91,35 +97,94 @@
...
+            <artifactId>exec-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>docker-prune</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${basedir}/../../.github/docker-prune.sh</executable>
```
Comment (Sanne) [inline integration-tests/elasticsearch-rest-client/pom.xml:179]: Interesting, I just learned about this script.

**Review by Sanne — APPROVED** [review]
Perfect, thanks!

File: integration-tests/elasticsearch-rest-client/pom.xml (hunk 1 above)
Comment (yrodiere) [reply integration-tests/elasticsearch-rest-client/pom.xml:179]: I copy-pasted that from another pom file. Not sure it's useful, but well... that's how it works everywhere else :shrug:

**Pull request merged**